c2rust-analyze/src/rewrite/expr/convert.rs

```diff
@@ -304,15 +347,19 @@ impl<'tcx> ConvertVisitor<'tcx> {
                         Rewrite::Let(vec![("byte_len".into(), self.get_subexpr(ex, 0))]),
                         Rewrite::Let1(
                             "n".into(),
-                            Box::new(format_rewrite!("byte_len as usize / {elem_size}")),
+                            Box::new(format_rewrite!(
+                                "byte_len as usize / std::mem::size_of::<{elem_ty}>()"
```

Comment:
Should we have a check here (maybe an assertion) that `byte_len` is a multiple of the element size?

Reply:
Added an assertion for this and similar cases: dfecf02